os/binfmt/binfmt_execmodule.c

```diff
@@ -291,6 +301,12 @@ int exec_module(FAR const struct binary_s *binp)
 	return (int)pid;
 
 errout_with_tcbinit:
+#ifdef CONFIG_FAULT_MGR
+	/* Deregister the application with from fault manager */
+
+	ret = fm_deregister_app(tcb->cmn.pid);
+	DEBUGASSERT(ret != ERROR);
```

Comment:
In this case, if it asserts, what will happen?

Reply:
It will be considered as kernel assert and board will be rebooted